src/twisted/trial/_asynctest.py

```diff
@@ -295,7 +299,9 @@ def _runFixturesAndTest(self, result):
         finally:
             self._undeprecateReactor(reactor)
 
-    def addCleanup(self, f, *args, **kwargs):
+    def addCleanup(
```

Comment:
this changes the defintion of addCleanup to match the TestCase.addCleanup - this is a breaking change where we should follow python or add `# type: ignore[override]`

Reply:
I can't see what's breaking about this change? It looks like it's just the annotation.

Reply:
It makes f a named only argument

Reply:
It changes the behaviour of `self.addCleanup(f=cleanup)`

Reply:
For reference. stdlib has `def addCleanup(self, function, /, *args, **kwargs):`

I think that is OK to make it positional only... to get it in sync with stdlib.

I think that this can qualify for an API exception.

if @glyph also agree with the breaking change, I think that we should have enough votes to pass the API exception.